crates/red_knot_python_semantic/src/types.rs

```diff
@@ -456,9 +501,6 @@ mod tests {
         );
     }
 
-    #[ignore = "\
-A spurious second 'Unresolved import' diagnostic message is emitted on `b.py`, \
-despite the symbol existing in the symbol table for `a.py`"]
```

Comment:
Following https://github.com/astral-sh/ruff/commit/ecd9e6a650ef428be67bb0e28cb0c52d27eb2895, the key benefit of this PR is now that this test can be unskipped. This improvement is also reflected in the fact that six spurious "unresolved import" diagnostics go away in the benchmarks.